BCStudentSoftwareDevTeam/celts
# Refactor Program Management section of Admin Settings

## Issue
Issue: #797

## Changes
- Created a modal to update the information displayed for the programs on the homepage.
- Fields include program name, description, partner, contact email, contact name, and location.
- Implemented a add a new file feature, to change the image of the programs on the homepage.
- Saved changes button works as expected, and there are two ways to close the modal, cancel or cross button.
- Partner, contact email, contact name, location all stored in database.
- Program name, description, images and links  can be directly seen in the home screen.
- Added editable link for Instagram, Facebook and Berea URL.
- Refactored testing function

## Testing
- Checkout `refactorProgramMgmt`
- `./database/reset_database.sh test`
- `flask run`
- Go to Admin -> Settings
- Open up the Program Settings accordion
- Click on 3-4 programs and upload a png to 2 and a jpg to 2.
- Go to the landing page and make sure the landing page images for each program reflects what you uploaded.
- Go to `app/static/files/programattachments` and make sure that there is only one file for each program (1.jpg, 2.png, etc.)
- Once your testing has concluded delete all files that were just added and are not source controlled (pngs or any other different file types)
- Run `git restore .`
- If it does not properly add back all files that were initially there you may have to manually add them from GitHub or an earlier commit.
